### ISSUE TITLE
self.cache_name would conflict with other variables

### ISSUE THREAD

**Issue by zhangrengang** [jackmaney/lazy-property#3]
For example:
```
import lazy_property

class OneClass(object):
    def __init__(self):
          self._name = None
    @lazy_property.LazyProperty
    def name(self):
        return True
```
`self.name` will return the value of `self._name`.

**Comment by jtv8** [comment]
I agree with @zhangrengang. This causes some non-intuitive behaviour: it's not clear to a casual reader of code written with LazyProperty that this is how it's implemented in the background, and the likelihood of these cached attributes being overwritten by other code is quite high. (It's happened twice to me just today!)

As an alternative, may I suggest using a dict to store cached property values on the parent object, using a dunder name like `__lazy_property_cache__`? That would be exceedingly unlikely to be overwritten by accident.
